back/app/serializers/web_api/v1/custom_field_serializer.rb

```diff
@@ -2,12 +2,21 @@
 
 class WebApi::V1::CustomFieldSerializer < WebApi::V1::BaseSerializer
   attributes :key, :input_type, :title_multiloc, :required, :ordering,
-    :enabled, :code, :created_at, :updated_at, :logic, :answer_visible_to
+    :enabled, :code, :created_at, :updated_at, :logic
 
   attribute :description_multiloc do |field|
     TextImageService.new.render_data_images field, :description_multiloc
   end
 
+  attribute :answer_visible_to, if: proc { |object, _params|
+    if object.resource && object.resource_type == 'CustomForm'
```

Comment:
This will cause N+1 query issues for the idea_custom_fields#index endpoint. Perhaps we should consider including the resource and participation context in the controller?

Reply:
The same will apply to the constraints, so worth moving it so it's only hit once